main.py

```diff
@@ -1,4 +1,280 @@
+import re
+import sqlite3
+from sqlite3 import Error
+import timeit
+
+
+# Class provides functionality to search for fb2 files
+class Fb2Reader:
+
+    @logger.catch()
+    def fb2_file_search(self):
+        os.chdir('./res/input')
+        self.__incorrect_file_removal()
+        fb2_file_list = {file for file in glob.glob("*.fb2")}
+        logger.info('File search is done')
+        return fb2_file_list
+
+    @logger.catch()
+    def parse_fb2(self, fb2_file):
+        try:
+            tree = ET.parse(fb2_file)
+            logger.info('File is parsed')
+        except Error as e:
+            logger.error('File is not parsed')
+        return tree
+
+    @logger.catch()
+    def __incorrect_file_removal(self):  # private
+        file_list = {file for file in glob.glob("*.*") if file not in glob.glob("*.fb2")}
+        path_parent = os.path.dirname(os.getcwd())
+        os.chdir(os.path.dirname(os.getcwd()))
+        if not os.path.exists('./incorrect_input'):
+            os.makedirs('./incorrect_input')
+        os.path.dirname(os.getcwd())
+        os.chdir('./input')
+        for f in file_list:
+            shutil.move(f, os.path.dirname(os.getcwd()) + '\\incorrect_input')
+            logger.debug(f + ' is moved to the incorrect_input folder')
+        logger.info('Incorrect files are removed')
+
+
+# Class provides functionality to analyse fb2 files
+class FileService:
+
+    @logger.catch()
+    def count_paragraph(self, elem_tree):
+        cnt_par_all = 0
+        # if we want to count all the paragraphs (all <p> elements)
+        for __ in elem_tree.iter('{http://www.gribuser.ru/xml/fictionbook/2.0}p'):
+            cnt_par_all += 1
+        return cnt_par_all
+
+    @logger.catch()
+    def count_words(self, elem_tree):
+        cnt_all_words = 0
+        # if we want to count all the words
+        for paragraphs in elem_tree.iter('{http://www.gribuser.ru/xml/fictionbook/2.0}p'):
+            if (paragraphs.text is None) or (len(paragraphs.text) == 0):
+                continue
+            cnt_all_words += len(paragraphs.text.split())
+        return cnt_all_words
+
+    @logger.catch()
+    def count_letters(self, elem_tree):
+        cnt_letters_all = 0
+        # if we want to count all the letters
+        for paragraphs in elem_tree.iter('{http://www.gribuser.ru/xml/fictionbook/2.0}p'):
+            if (paragraphs.text is None) or (len(paragraphs.text) == 0):
+                continue
+            for word in paragraphs.text.split():
+                cnt_letters_all += len(word)
+
+        return cnt_letters_all
+
+    @logger.catch()
+    def count_words_with_capital_letters(self, elem_tree):
+        cnt_cap_letters_all = 0
+
+        for paragraphs in elem_tree.iter('{http://www.gribuser.ru/xml/fictionbook/2.0}p'):
+            if (paragraphs.text is None) or (len(paragraphs.text) == 0):
+                continue
+            for word in paragraphs.text.split():
+                if word == word.capitalize():
```

Comment:
Эта проверка ложно срабатывает на цифры и символы. Было бы лучше написать:

```suggestion
                if word[0].isupper():
```

![image](https://user-images.githubusercontent.com/34323554/97810301-93228400-1c7b-11eb-907e-0da102b32c4b.png)

Reply:
О, отличное решение. Сразу вообще Isupper и использовала, но потом поняла, что большой должна быть именно первая буква, а не все слово и нашла capitalize команду. Не подумала, что можно же взять просто первый символ строки)